server/events/project_finder.go

```diff
@@ -168,6 +181,53 @@ func (p *DefaultProjectFinder) filterToFileList(log logging.SimpleLogging, files
 	return filtered
 }
 
+func (p *DefaultProjectFinder) FindMatchingProjects(log logging.SimpleLogging, repo models.Repo, pull models.PullRequest) ([]string, error) {
+	var pullInfoList []string
+	repoDir, err := p.WorkingDir.GetWorkingDir(repo, pull, "default")
+	if err != nil {
+		return pullInfoList, errors.Wrap(err, "retrieving working dir")
+	}
+
+	modifiedFiles, err := p.VCSClient.GetModifiedFiles(repo, pull)
+	if err != nil {
+		return pullInfoList, errors.Wrap(err, "retrieving modified files")
+	}
+
+	hasRepoCfg, err := p.ParserVarlidator.HasRepoCfg(repoDir)
```

Comment:
Hmm I'm not sure this is the best approach to get the list of projects for a given PR.  I think you can just leverage the db here:
https://github.com/lyft/atlantis/blob/release-v0.17.1-lyft.1/server/events/db/boltdb.go#L380

It should give you all the projects in a given pull request. Then you shouldn't need this whole function at all.

Reply:
+1 on using `Db.GetPullStatus(pull)`.